google/generativeai/text.py

```diff
@@ -49,6 +61,34 @@ def _make_generate_text_request(
     safety_settings: safety_types.SafetySettingOptions | None = None,
     stop_sequences: str | Iterable[str] | None = None,
 ) -> glm.GenerateTextRequest:
+    """
+    Creates a GenerateTextRequest object based on the provided parameters.
+
+    This function generates a glm.GenerateTextRequest object with the specified
+    parameters. It prepares the input parameters and creates a request that can be
+    used for generating text using the chosen model.
+
+    Args:
+        model (model_types.ModelNameOptions, optional): The model to use for text generation.
```

Comment:
I'd prefer to avoid repeating the type information from the function signature in docstring. That just lets things slip out of sync.